src/pages/instances/forms/UploadExternalFormatFileForm.tsx

```diff
@@ -0,0 +1,346 @@
+      },
+    });
+  };
+
+  const handleSubmit = (values: UploadExternalFormatFileFormValues) => {
+    if (!values.imageFile) {
+      return;
+    }
+
+    // start create instance operation
+    createInstance(uploadExternalFormatFilePayload(values), project?.name || "")
+      .then((operation) => {
+        const operationId = operation.metadata.id;
+        const operationSecret = operation.metadata?.metadata?.fs;
+        const instanceName = getInstanceName(operation.metadata);
+
+        // establish websocket connection based on the instance creation operation
+        let wsUrl = `wss://${location.host}/1.0/operations/${operationId}/websocket`;
+        if (operationSecret) {
+          wsUrl += `?secret=${operationSecret}`;
+        }
+
+        sendFileByWebSocket(
+          wsUrl,
+          formik.values.imageFile,
+          handleUploadProgress.bind(null, instanceName),
+          handleUploadError,
+        );
+
+        // set up event queue for the operation
+        eventQueue.set(
+          operationId,
+          () => handleSuccess(getInstanceName(operation.metadata)),
+          handleFailure,
+          invalidateCache,
+        );
+      })
+      .catch((e) => {
+        notify.failure("Instance creation failed.", e);
+        formik.setSubmitting(false);
+        setUploadState(null);
+      });
+  };
+
+  const archOptions = supportedVMArchOptions(
+    settings?.environment?.architectures || [],
+  );
+
+  const formik = useFormik<UploadExternalFormatFileFormValues>({
+    initialValues: {
+      name: "",
+      pool: "",
+      imageFile: null,
+      formatConversion: true,
+      virtioConversion: false,
+      architecture: archOptions[0]?.value,
+    },
+    validateOnMount: true,
+    validationSchema: Yup.object().shape({
+      name: instanceNameValidation(
+        project?.name || "",
+        instanceNameAbort,
+      ).optional(),
+    }),
+    onSubmit: handleSubmit,
+  });
+
+  const handleFileChange = async (e: ChangeEvent<HTMLInputElement>) => {
+    const { onChange } = formik.getFieldProps("imageFile");
+    onChange(e);
+
+    if (e.currentTarget.files) {
+      const file = e.currentTarget.files[0];
+      const instanceName = fileToInstanceName(file.name, "-import");
+      await formik.setFieldValue("imageFile", file);
+      await formik.setFieldValue("name", instanceName);
+
+      // validate instance name
+      await formik.validateField("name");
+      void formik.setFieldTouched("name", true, true);
+      if (!formik.errors.name) {
+        formik.setFieldError("name", undefined);
+      }
+
+      // If the image is already in raw format, remove the format conversion option
+      // this will optimise the conversion process since raw vm images do not need to be converted
+      const isRawImage = await isImageTypeRaw(file);
+      if (isRawImage) {
+        await formik.setFieldValue("formatConversion", !isRawImage);
+      }
```

Comment:
```suggestion
      await formik.setFieldValue("formatConversion", !isRawImage);
```
I think both cases are important, for non-raw images we should also enable conversion. This is the default, but a user might change between raw and non-raw files.

Reply:
of course, that was my bad for not removing the condition check